### ISSUE TITLE
TypeError: Cannot read property 'restart' of null

### ISSUE THREAD

**Issue by BrightDev1110** [avishayil/react-native-restart#249]
React native : 0.72.4
 react-native-restart: 0.0.27
 
 
 WARN  Possible Unhandled Promise Rejection (id: 0):
TypeError: Cannot read property 'restart' of null
TypeError: Cannot read property 'restart' of null

**Comment by spideyinf** [comment]
I do have the same issue, Has anyone with any solution helped?

**Comment by bviebahn** [comment]
I also had this issue earlier, but after restarting the bundler and rebuilding the app it went away.

**Comment by master12** [comment]
Have the same issue. Rebuilding the bundle did not help.

**Comment by bilalaurangzebkhan** [comment]
To date, I'm in the same issue if anyone finds a solution please help to resolve

**Comment by bilalaurangzebkhan** [comment]
Make sure rebuilding your app after installing react-native-restart **_It will not work in Expo Go_**

You can create a development build which is like a custom version of Expo Go. _**It won't work in Expo Go, because it includes native code,**_ and that native code is not built into Expo Go. (And can't be added later.) So you need to build your app. e.g.:

`eas build -p android --profile development`

Then install that on your device/emulator and use it **instead of Expo Go**.

**Comment by bilalaurangzebkhan** [comment]
> > Make sure rebuilding your app after installing react-native-restart **_It will not work in Expo Go_**
> > You can create a development build which is like a custom version of Expo Go. _**It won't work in Expo Go, because it includes native code,**_ and that native code is not built into Expo Go. (And can't be added later.) So you need to build your app. e.g.:
> > `eas build -p android --profile development`
> > Then install that on your device/emulator and use it **instead of Expo Go**.
> 
> Thanks @bilalaurangzebkhan. This explanation should be in README.. Does this mean if we install it via npm / yarn add react-native-restart, create a dev-build, that's it? No need for any of that linking stuff?

Yes @ansmlc the newer version does auto-linking. and the react-native-restart will not work on expo Go but will work fine for development and production build.

**Comment by avishayil** [comment]
This error indicates the native module isn't properly linked. Please try these steps:

1. Reinstall the module:
   npm install react-native-restart@latest
   
2. iOS:
   cd ios
   pod install
   cd ..
   npx react-native run-ios --clean
   
3. Android:
   cd android
   ./gradlew clean
   cd ..
   npx react-native run-android

4. Verify the import:
   ```javascript
   import RNRestart from 'react-native-restart';
   console.log('RNRestart module:', RNRestart); // Should not be null
   ```

If still failing, check:
- Your React Native version supports autolinking (>= 0.60)
- No conflicting versions in node_modules
- The native module built successfully (check build logs)

Closing - please reopen with full error logs if the issue persists.
